src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/Models/ReindexJobRecord.cs

```diff
@@ -31,13 +31,17 @@ public class ReindexJobRecord : JobRecord
         public ReindexJobRecord(
             IReadOnlyDictionary<string, string> searchParametersHash,
             IReadOnlyCollection<string> targetResourceTypes,
+            IReadOnlyCollection<string> targetSearchParameterTypes,
+            IReadOnlyCollection<string> searchParameterResourceTypes,
             ushort maxiumumConcurrency = 1,
             uint maxResourcesPerQuery = 100,
             int queryDelayIntervalInMilliseconds = 500,
             ushort? targetDataStoreUsagePercentage = null)
         {
-            EnsureArg.IsNotNull(searchParametersHash, nameof(searchParametersHash));
-            EnsureArg.IsNotNull(targetResourceTypes, nameof(targetResourceTypes));
+            ResourceTypeSearchParameterHashMap = EnsureArg.IsNotNull(searchParametersHash, nameof(searchParametersHash));
+            TargetResourceTypes = EnsureArg.IsNotNull(targetResourceTypes, nameof(targetResourceTypes));
+            TargetSearchParameterTypes = EnsureArg.IsNotNull(targetSearchParameterTypes, nameof(targetSearchParameterTypes));
```

Comment:
As these are optional parameters for the user, I assume we check for not null because they are always initialized to an empty collection?